### PR TITLE
Auto select last team viewed

### PR DESCRIPTION
**Pull request by csduarte**

### CODE REVIEW

**Comment by thomchop** [comment]
@jarredwitt In my comment on #162, I encouraged the amending of WIP commits. I see that you did that here with my initial WIP commit. Great!
But when you went to push the amended commit, the push was rejected. It's because your local branch and the remote branch diverged at that point. The resulting changes were still identical, but the commits had different hashes/SHAs.
When you pushed, git probably output something like this:
```
$ git push
To https://github.com/csduarte/mattermost-mobile.git
 ! [rejected]        auto-select-last-team-viewed -> auto-select-last-team-viewed (non-fast-forward)
error: failed to push some refs to 'https://github.com/csduarte/mattermost-mobile.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```
Git is correct that you need to `Integrate the remote changes before pushing again`, but when it says `e.g. 'git pull ...'`, don't listen to it! Or at least don't blindly follow it. This is one of the gotchas that virtually ever git user out there has been lead astray by.
This warning should always be heeded (you don't want to blindly force push), but you've got to determine the right step to take next based on the situation. In this case, you definitely wanted to force push. Here are the criteria for that:
* you're pushing to your own branch
* nobody else is depending on this branch (for another PR, fork, release, deploy, etc)
* the remote branch doesn't contain any truly new commits that you're missing locally
* the divergence was caused by a rebase/amend that you did intentionally
If all of the above are true then this is a clear "force push" scenario.
The last criterium is particularly important. You always want to confirm that you know why your push was rejected. If the warning is unexpected then it might indicate you've inadvertently rewritten your branch's git history.

File: app/scenes/root/root_container.js [hunk 1]
```diff
@@ -16,13 +17,17 @@ function mapStateToProps(state, ownProps) {
         ...ownProps,
         credentials: state.entities.general.credentials,
         logoutRequest: state.requests.users.logout,
-        loginRequest: state.requests.users.login
+        loginRequest: state.requests.users.login,
+        currentTeamId: state.entities.teams.currentId,
+        teams: state.entities.teams.teams
```
Comment (enahum) [inline app/scenes/root/root_container.js]: for currentTeamId and teams we could use the **selectors** in `service -> selectors -> teams`

File: app/scenes/root/root_container.js [hunk 2]
```diff
@@ -16,13 +17,17 @@ function mapStateToProps(state, ownProps) {
...
     };
 }
 
 function mapDispatchToProps(dispatch) {
     return {
         actions: bindActionCreators({
+            ...teamActions,
```
Comment (enahum) [inline app/scenes/root/root_container.js]: do we need all the **teamActions**? cause actually I don't see any of them being used in the `root` component

File: app/scenes/select_team/select_team_container.js [hunk 3]
```diff
@@ -23,7 +25,8 @@ function mapDispatchToProps(dispatch) {
     return {
         actions: bindActionCreators({
             goToChannelView,
-            selectTeam,
+            ...storageActions,
+            ...teamActions,
```
Comment (enahum) [inline app/scenes/select_team/select_team_container.js]: do we need **all** the actions in team and storage?

File: app/scenes/root/root_container.js (hunk 1 above)
Comment (enahum) [inline app/scenes/root/root_container.js]: What about the selectors?

**Pull request merged**